code/menuui/techmenu.cpp

```diff
@@ -896,35 +895,43 @@ void techroom_change_tab(int num)
 		case INTEL_DATA_TAB:
 
 			// load intel if necessary
-			if ( Intel_loaded == 0 ) {
-				// now populate the entry structs
+			if ( !Intel_loaded ) {
+				if (Intel_list == nullptr) {
+					Intel_list = new tech_list_entry[Intel_info.size()];
+
+					if (Intel_list == nullptr)
```

Comment:
This will never be true since a failure to allocate the data with throw a `std::bad_alloc` exception. You could ass `std::nothrow` to the `new` but I don't really see the point in handling that since we always assume to have enough memory.

Reply:
Good point.  I'll change this.